libraries/botframework-connector/tests/auth/managedIdentityServiceClientCredentialsFactory.test.js

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+const assert = require('assert');
+const {
+    ManagedIdentityServiceClientCredentialsFactory,
+} = require('../../lib/auth/managedIdentityServiceClientCredentialsFactory');
+const { JwtTokenProviderFactory } = require('../../lib/auth/jwtTokenProviderFactory');
+
+const testAppId = 'foo';
+const testAudience = 'bar';
+const testTokenProvider = new JwtTokenProviderFactory(testAppId);
+
+const testInput = [
+    { name: 'null', value: null },
+    { name: 'empty', value: '' },
+    { name: 'blank', value: ' ' },
+];
+const errorMsgAppId = 'ManagedIdentityServiceClientCredentialsFactory.constructor(): missing appid.';
+const errorMsgToken = 'ManagedIdentityServiceClientCredentialsFactory.constructor(): missing tokenProviderFactory.';
+
+describe('ManagedIdentityServiceClientCredentialsFactory', function () {
+    describe('Constructor', function () {
+        it('succeeds with appId and tokenProvider', async function () {
+            const sut = new ManagedIdentityServiceClientCredentialsFactory(testAppId, testTokenProvider);
+            assert.strictEqual(testAppId, sut.appId);
+            assert.strictEqual(testTokenProvider, sut.tokenProviderFactory);
+        });
+
+        testInput.forEach(({ name, value }) => {
+            it(`should throw if appId is ${name}`, function () {
+                assert.throws(() => new ManagedIdentityServiceClientCredentialsFactory(value, testTokenProvider), {
+                    name: 'AssertionError',
+                    message: errorMsgAppId,
+                });
+            });
+        });
+
+        it('should throw with null tokenProviderFactory', function () {
+            assert.throws(() => new ManagedIdentityServiceClientCredentialsFactory(testAppId, null), {
+                name: 'AssertionError',
+                message: errorMsgToken,
+            });
+        });
+    });
+
+    describe('IsValidAppId', function () {
+        it('should be valid', async function () {
+            const sut = new ManagedIdentityServiceClientCredentialsFactory(testAppId, testTokenProvider);
+            assert.strictEqual(true, await sut.isValidAppId(testAppId));
+        });
+
+        it('should not be valid', async function () {
+            const sut = new ManagedIdentityServiceClientCredentialsFactory(testAppId, testTokenProvider);
+            assert.strictEqual(false, await sut.isValidAppId('invalidAppId'));
+        });
+    });
+
+    describe('IsAuthenticationDisabled', function () {
+        it('should not be disabled', async function () {
+            const sut = new ManagedIdentityServiceClientCredentialsFactory(testAppId, testTokenProvider);
+            assert.strictEqual(false, await sut.isAuthenticationDisabled());
+        });
+    });
```

Comment:
Please see my earlier remarks for collapsing the `describe()` and `it()` together. [passwordServiceClientCredentialFactory.test.js](https://github.com/microsoft/botbuilder-js/blob/9c9f8d95927ea2f035af97ba53ddaa73e0231289/libraries/botframework-connector/tests/passwordServiceClientCredentialFactory.test.js#L17-L28) is a good reference point.